src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -284,6 +312,60 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructors(
         return email;
     }
     
+    private List<EmailWrapper> generateFeedbackSessionClosedEmail(
+            CourseAttributes course, FeedbackSessionAttributes session,
+            List<InstructorAttributes> instructors, List<StudentAttributes> students) {
+
+        List<EmailWrapper> emails = new ArrayList<EmailWrapper>();
+
+        for (InstructorAttributes instructor : instructors) {
+            emails.add(feedbackSessionClosedEmailForInstructor(course, session, instructor));
+        }
+        for (StudentAttributes student : students) {
+            emails.add(feedbackSessionClosedEmailForStudent(course, session, student));
+        }
+        
+        return emails;
+    }
+    
+    private EmailWrapper feedbackSessionClosedEmailForInstructor(CourseAttributes course, FeedbackSessionAttributes session,
+            InstructorAttributes instructor) {
+        String template = EmailTemplates.USER_FEEDBACK_SESSION_CLOSED;
+        String subject = EmailType.FEEDBACK_CLOSED.getSubject();
+
+        String emailBody = Templates.populateTemplate(template,
+                "${userName}", instructor.name,
+                "${courseName}", course.getName(),
+                "${courseId}", course.getId(),
+                "${feedbackSessionName}", session.getFeedbackSessionName(),
+                "${deadline}", TimeHelper.formatTime12H(session.getEndTime()),
+                "${supportEmail}", Config.SUPPORT_EMAIL);
+
+        EmailWrapper email = getEmptyEmailAddressedToEmail(instructor.email);
+        email.setSubject(String.format(subject, course.getName(), session.getFeedbackSessionName()));
+        email.setContent(emailBody);
+        return email;
+    }
+    
+    private EmailWrapper feedbackSessionClosedEmailForStudent(CourseAttributes course, FeedbackSessionAttributes session,
+            StudentAttributes student) {
+        String template = EmailTemplates.USER_FEEDBACK_SESSION_CLOSED;
+        String subject = EmailType.FEEDBACK_CLOSED.getSubject();
+
+        String emailBody = Templates.populateTemplate(template,
+                "${userName}", student.name,
+                "${courseName}", course.getName(),
+                "${courseId}", course.getId(),
+                "${feedbackSessionName}", session.getFeedbackSessionName(),
+                "${deadline}", TimeHelper.formatTime12H(session.getEndTime()),
+                "${supportEmail}", Config.SUPPORT_EMAIL);
+
+        EmailWrapper email = getEmptyEmailAddressedToEmail(student.email);
+        email.setSubject(String.format(subject, course.getName(), session.getFeedbackSessionName()));
+        email.setContent(emailBody);
+        return email;
```

Comment:
This is just extracting the two similar routines to two similar methods. While it has its own advantage of increasing SLAP, it doesn't address that both methods are almost identical. You should make a method (let's say Method A) that takes in `course, session, userName, userEmail` and make these two methods call Method A. It's also fine if you want the `generateFeedbackSessionClosedEmail` to call Method A directly.